scipy/interpolate/_polyint.py

```diff
@@ -589,21 +583,72 @@ class BarycentricInterpolator(_Interpolator1DWithDerivatives):
 
     Notes
     -----
-    This class uses a "barycentric interpolation" method that treats
-    the problem as a special case of rational function interpolation.
-    This algorithm is quite stable, numerically, but even in a world of
-    exact computation, unless the x coordinates are chosen very
-    carefully - Chebyshev zeros (e.g., cos(i*pi/n)) are a good choice -
-    polynomial interpolation itself is a very ill-conditioned process
-    due to the Runge phenomenon.
+    This method is a variant of Lagrange polynomial interpolation [1]_ based on [2]_.
+    Instead of using Lagrange's or Newton's formula, the polynomial is represented by
+    the barycentric formula
+
+    .. math::
+
+        p(x) =
+        \frac{\sum_{i=1}^m\ w_i y_i / (x - x_i)}{\sum_{j=1}^m w_i / (x - x_i)},
+
+    where :math:`w_i` are the barycentric weights computed with to the general
+    formula
+
+    .. math::
+
+        w_i = \left( \prod_{k \neq i} x_i - x_k \right)^{-1}.
+
+    This is the same barycentric form used by `AAA` and `FloaterHormannInterpolator`.
+    However, in contrast, the weights :math:`w_i` are defined such that
+    :math:`p(x)` is a polynomial rather than a rational function.
+
+    The barycentric representation avoids many of the problems associated with
+    polynomial interpolation caused by floating-point arithmetic. However, it does not
+    avoid issues that are intrinsic to polynomial interpolation. Namely, if the
+    x-coordinates `xi` are equally spaced then the weights `wi` can be computed
```

Comment:
```suggestion
    x-coordinates `xi` are equally spaced, then the weights `wi` can be computed
```

Reply:
This was probably fine; I actually forgot that the inputs were named `xi` and `wi`.

Reply:
I decided to change it anyway since it seemed odd for the notation to jump around